server/modules/search/db/engine.js

```diff
@@ -52,6 +52,7 @@ module.exports = {
         })
       })
       .limit(WIKI.config.search.maxHits)
+    results = results.filter(page => WIKI.auth.checkAccess(user, ['read:pages'], page))
```

Comment:
Move the access check outside of the search engine module, into the page resolver instead. Otherwise this logic would be duplicated in every search module which is not necessary.

Reply:
sorry im late on this, where / what is the page resolver?

Reply:
Wouldn't this be good for searching through the wiki and hiding results that the user doesn't have permission for?